Documentation/ExtensionArchitecture/Extbase/Reference/FileUpload.rst

```diff
@@ -0,0 +1,489 @@
+
+Configuration options for file uploads
+--------------------------------------
+
+The configuration for a file upload is defined in a
+:php:`FileUploadConfiguration` object.
+
+This object contains the following configuration options.
+
+..  hint::
+
+    The appropriate setter methods or configuration
+    keys can best be inspected inside that class definition.
+
+..  _extbase_fileupload_attribute-property-name:
+
+Property name:
+~~~~~~~~~~~~~~
+
+Defines the name of the property of a domain model to which the file upload
+configuration applies. The value is automatically retrieved when using
+the :php-short:`\TYPO3\CMS\Extbase\Annotation\FileUpload` attribute. If the
+:php-short:`\TYPO3\CMS\Extbase\Mvc\Controller\FileUploadConfiguration` object
+is created manually, it must be set using the :php:`$propertyName`
+constructor argument.
+
+..  _extbase_fileupload_attribute-validation:
+
+Validation:
+~~~~~~~~~~~
+
+File upload validation is defined in an array of validators in the
+:php-short:`\TYPO3\CMS\Extbase\Mvc\Controller\FileUploadConfiguration` object. The validator
+:php:`\TYPO3\CMS\Extbase\Validation\Validator\FileNameValidator`,
+which ensures that no executable PHP files can
+be uploaded, is added by default if the file upload configuration object
+is created using the
+:php-short:`\TYPO3\CMS\Extbase\Annotation\FileUpload` attribute.
+
+In addition, Extbase includes the following validators to validate an
+:php-short:`\TYPO3\CMS\Core\Http\UploadedFile` object:
+
+*   :php:`\TYPO3\CMS\Extbase\Validation\Validator\FileSizeValidator`
+*   :php:`\TYPO3\CMS\Extbase\Validation\Validator\MimeTypeValidator`
+*   :php:`\TYPO3\CMS\Extbase\Validation\Validator\ImageDimensionsValidator`
+
+Those validators can either be configured with the
+:php-short:`\TYPO3\CMS\Extbase\Annotation\FileUpload` attribute or added
+manually to the configuration object
+with the :php:`addValidator` method.
+
+..  _extbase_fileupload_attribute-required:
+
+Required:
+~~~~~~~~~
+
+Defines whether a file must be uploaded. If it is set to `true`, the
+:php:`minFiles` configuration is set to `1`.
+
+..  _extbase_fileupload_attribute-minimum-files:
+
+Minimum files:
+~~~~~~~~~~~~~~
+
+Defines the minimum amount of files to be uploaded.
+
+..  _extbase_fileupload_attribute-maximum-files:
+
+Maximum files:
+~~~~~~~~~~~~~~
+
+Defines the maximum amount of files to be uploaded.
+
+..  _extbase_fileupload_attribute-upload-folder:
+
+Upload folder:
+~~~~~~~~~~~~~~
+
+Defines the upload path for the file upload. This configuration expects a
+storage identifier (e.g. :php:`1:/user_upload/folder/`). If the given target
+folder in the storage does not exist, it is created automatically.
+
+..  _extbase_fileupload_attribute-upload-folder-creation:
+
+Upload folder creation, when missing:
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The default creation of a missing storage folder can be disabled via the
+configuration attribute :php:`createUploadFolderIfNotExist`
```

Comment:
What is a "configuration attribute"?

Reply:
The attribute specifies an array in the #[FileUpload([....])] block; a key inside there is what this refers to.

Maybe "attribute configuration array key" is better? :D

Reply:
It is a property in an attribute, not a property itself. Maybe "via the property createUploadFolderIfNotExist" in the FileUpload attribute"?

Reply:
done